aloha-core/src/main/scala/com/eharmony/aloha/dataset/csv/CsvColumnarRowCreator.scala

```diff
@@ -0,0 +1,169 @@
+package com.eharmony.aloha.dataset.csv
+
+import com.eharmony.aloha.dataset._
+import com.eharmony.aloha.dataset.csv.encoding.{Encoding, HotOneEncoding, RegularEncoding, ThermometerEncoding}
+import com.eharmony.aloha.dataset.csv.finalizer.{BasicColumnarFinalizer, EncodingBasedColumnarFinalizer}
+import com.eharmony.aloha.dataset.csv.json._
+import com.eharmony.aloha.dataset.csv.CsvColumnarRowCreator.{ColumnName, ColumnType}
+import com.eharmony.aloha.reflect.{RefInfo, RefInfoOps}
+import com.eharmony.aloha.semantics.compiled.CompiledSemantics
+import com.eharmony.aloha.semantics.func.GenAggFunc
+import spray.json.JsValue
+
+import scala.collection.{breakOut, immutable => sci}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.{Failure, Success, Try}
+
+/**
+  * A `CsvColumnarRowCreator` makes a sequence of `String`-based column values.
+  *
+  * '''NOTE''': string representations of column types are the simple name for values extending
+  * `AnyVal`.  For instance: `"Byte"`, `"Char"`, `"Short"`, `"Int"`, `"Long"`, `"Float"`, `"Double"`
+  * and non-`AnyVal`s are the canonical class name.  For instance, Strings are the represented as
+  * `"java.lang.String"`.
+  *
+  * Created by ryan.deak on 2/27/18.
+  * @param features a feature extractor function.
+  * @param headersToTypes An ordering mapping from column name to type.  The size should be the
+  *                       output size of the `apply` function.
+  * @param nullString the string representing null.
+  * @tparam A The domain of the row creator.
+  */
+final case class CsvColumnarRowCreator[-A] private[csv](
+    features: FeatureExtractorFunction[A, Seq[String]],
+    headersToTypes: Seq[(ColumnName, ColumnType)],
+    nullString: String
+) extends RowCreator[A, sci.IndexedSeq[String]] {
+
+  /**
+    * Column names
+    * @return
+    */
+  def headers: Seq[String] = headersToTypes.unzip._1
```

Comment:
This is called infrequently.  Maybe once or twice.  For instance, it could be called once to construct a `Encoder[Row]` in spark to get the column names.